### PR TITLE
Propagate swift version / copts

### PR DESCRIPTION
**Pull request by rahul-malik**

### CODE REVIEW

**Comment by jerrymarino** [comment]
@rahul-malik - Awesome! This seems to be causing a CI issue:
```In file included from /Users/travis/build/pinterest/xchammer/IntegrationTests/Sandbox/UrlGet/Vendor/Stripe/Stripe/NSError+Stripe.m:9:
/Users/travis/build/pinterest/xchammer/IntegrationTests/Sandbox/UrlGet/Vendor/Stripe/Stripe/NSError+Stripe.h:11:9: fatal error: 'StripeError.h' file not found
#import "StripeError.h"
        ^~~~~~~~~~~~~~~
1 error generated.
```

It's not immediately clear why..

**Comment by jerrymarino** [comment]
@rahul-malik it'd also be good to get a sample project with swift in it, so we can make sure everything is working correctly.

File: Sources/XCHammer/XCBuildSettings.swift [hunk 1]
```diff
@@ -161,6 +161,8 @@ struct XCBuildSettings: Encodable {
     var moduleMapFile: First<String>?
     // Disable Xcode derived headermaps, be explicit to avoid divergence
     var useHeaderMap: First<String>? = First("NO")
+    var swiftVersion: First<String>?
```
Comment (jerrymarino) [inline Sources/XCHammer/XCBuildSettings.swift]: @rahul-malik is this the correct data type in Xcode?

File: Sources/XCHammer/XCBuildSettings.swift [hunk 2]
```diff
@@ -211,6 +215,7 @@ struct XCBuildSettings: Encodable {
         var container = encoder.container(keyedBy: CodingKeys.self)
 
         try container.encode(copts.joined(separator: " "), forKey: .copts)
+        try container.encode(swiftCopts.joined(separator: " "), forKey: .copts)
```
Comment (jerrymarino) [inline Sources/XCHammer/XCBuildSettings.swift]: @rahul-malik did you intend to set `OTHER_SWIFT_FLAGS` here? This LOC overwrites `OTHER_CFLAGS`

File: Sources/XCHammer/XcodeTarget.swift [hunk 3]
```diff
@@ -593,6 +595,31 @@ public class XcodeTarget: Hashable, Equatable {
...
+                }
+                settings.swiftVersion <>=  First(swiftVersion)
+            case .has_swift_dependency, .has_swift_info:
+                guard let swiftVersion = try? ShellOut.shellOut(to: "xcrun swift package tools-version") else {
+                    break
+                }
+                settings.swiftVersion <>=  First(swiftVersion)
+            case .swiftc_opts:
+                if let coptsArray = value as? [String] {
+                    let processedOpts = coptsArray.map { opt -> String in
```
Comment (jerrymarino) [inline Sources/XCHammer/XcodeTarget.swift]: Do we need to patch these for `bazel-genfiles` ( and for `external` in a followup ). We should have an example of a `swift` genfile in a `swift_library` in a sample. Consider either adding this to `UrlGet` or just creating a new project where we can add all the swift edge cases.

File: Sources/XCHammer/XCBuildSettings.swift (hunk 1 above)
Comment (rahul-malik) [reply Sources/XCHammer/XCBuildSettings.swift]: yep it is!

File: Sources/XCHammer/XcodeTarget.swift (hunk 3 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: Probably patch this for external with the PR that handles external repos. Adding a new example is a good idea

**Comment by jerrymarino** [comment]
Hey @rahul-malik - this one LGTM, mostly. Would you add an example with some `.swift` files? To create a new sample and test, the `Frankenstein` app is a pretty minimal starting point.. It'd be good to exemplify this into a sample where we can add all relevant swift cases.

File: Sources/XCHammer/XcodeTarget.swift (hunk 3 above)
Comment (rahul-malik) [reply Sources/XCHammer/XcodeTarget.swift]: lets do this in a follow up PR

File: Sources/XCHammer/XcodeTarget.swift [hunk 4]
```diff
@@ -601,6 +602,31 @@ public class XcodeTarget: Hashable, Equatable {
                 break // These attrs are not related to XCConfigs
             case .binary:
                 break // Explicitly not handled since it is a implicit target we don't intend to handle
+            case .swift_language_version:
+                guard let swiftVersion = value as? String else {
+                    break
+                }
+                settings.swiftVersion <>=  First(swiftVersion)
+            case .has_swift_dependency, .has_swift_info:
+                guard let swiftVersion = try? ShellOut.shellOut(to: "xcrun swift package tools-version") else {
```
Comment (kastiglione) [inline Sources/XCHammer/XcodeTarget.swift]: This only works if there's a Package.swift, which not all projects will have. I've opened an issue: #124

**Pull request merged**